src/status_im/ui/screens/chat/message/pinned_message.cljs

```diff
@@ -0,0 +1,88 @@
+(ns status-im.ui.screens.chat.message.pinned-message
+  (:require [re-frame.core :as re-frame]
+            [status-im.i18n.i18n :as i18n]
+            [status-im.ui.components.colors :as colors]
+            [status-im.ui.components.react :as react]
+            [quo.core :as quo]
+            [reagent.core :as reagent]
+            [status-im.chat.models.pin-message :as models.pin-message]
+            [status-im.ui.components.list.views :as list]
+            [status-im.ui.components.radio :as radio]
+            [status-im.utils.handlers :refer [<sub]]
+            [status-im.ui.screens.chat.message.message :as message]))
+
+(def selected-unpin (reagent/atom nil))
+
+(defn render-pin-fn [{:keys [message-id outgoing] :as message}
+                     _
+                     _
+                     {:keys [group-chat public? current-public-key space-keeper]}]
+  [react/touchable-without-feedback {:style {:width "100%"}
+                                     :on-press #(reset! selected-unpin message-id)}
+   [react/view {:style {:flex-direction  :row
+                        :align-items     :center
+                        :justify-content :space-between
+                        :flex            1
+                        :padding-right   20}}
+    [message/chat-message
+     (assoc message
+            :group-chat group-chat
+            :public? public?
+            :current-public-key current-public-key
+            :show-input? false
+            :pinned false
+            :display-username? (not outgoing)
+            :display-photo? false
+            :last-in-group? false
+            :in-popover? true)
+     space-keeper]
+    [react/view {:style {:position    :absolute
+                         :right       18
+                         :padding-top 4}}
+     [radio/radio (= @selected-unpin message-id)]]]])
+
+(def list-key-fn #(or (:message-id %) (:value %)))
+
+(defn pinned-messages-view [chat-id]
```

Comment:
just wondering, where is this used? i see we have same function in pinned_messages.cljs

Reply:
it is used for `pin-limit-popover` popover where the user should pick a pinned message to unpin in order to pin a new one. but you are right i'll rename this as is confusing